File: BackEnd/src/domain/interfaces/usecases/community/IGetPostCommentsUseCase.ts
/**
 * @file IGetPostCommentsUseCase.ts
 * @description Interfaz del caso de uso para obtener comentarios de un post.
 * @module Community
 * @layer Domain
 */

import type { CommentResponseDTO } from '../../../dtos/community/comment-response.dto.js';

/**
 * Contrato del use case GetPostComments.
 */
export interface IGetPostCommentsUseCase {
  /**
   * Ejecuta el caso de uso.
   *
   * @param postId — Id del post cuyos comentarios se solicitan.
   * @returns Lista de CommentResponseDTOs ordenada por fecha ascendente.
   */
  execute(postId: string): Promise<CommentResponseDTO[]>;
}

File: FrontEnd/lib/domain/interfaces/usecases/community/i_get_post_comments_use_case.dart
/// @file i_get_post_comments_use_case.dart
/// @description Interfaz: Obtiene los comentarios de un post.
/// @module Community
/// @layer Domain
library;

import '../../../entities/comment.dart';

abstract interface class IGetPostCommentsUseCase {
  /// Obtiene los comentarios de un post ordenados por fecha de creación.
  ///
  /// [param] postId — Identificador del post.
  /// [returns] Lista de [Comment].
  Future<List<Comment>> execute(String postId);
}
